pkg/networkserver/downlink.go

```diff
@@ -193,7 +194,23 @@ outer:
 		down, dev.QueuedApplicationDownlinks = dev.QueuedApplicationDownlinks[0], dev.QueuedApplicationDownlinks[1:]
 
 		if len(down.FRMPayload) > int(maxDownLen) {
-			logger.Warn("Application downlink present, but the payload is too long, skipping...")
+			logger.Warn("Application downlink present, but the payload is too long, informing Application Server...")
+			ok, err := ns.handleASUplink(ctx, dev.EndDeviceIdentifiers.ApplicationIdentifiers, &ttnpb.ApplicationUp{
+				EndDeviceIdentifiers: dev.EndDeviceIdentifiers,
+				CorrelationIDs:       append(events.CorrelationIDsFromContext(ctx), down.CorrelationIDs...),
+				Up: &ttnpb.ApplicationUp_DownlinkFailed{
+					DownlinkFailed: &ttnpb.ApplicationDownlinkFailed{
+						ApplicationDownlink: *down,
+						Error:               *errors.ErrorDetailsToProto(errApplicationDownlinkTooLong).(*ttnpb.ErrorDetails),
```

Comment:
Why?

Reply:
Unfortunately it's a compile-time error without the assertion: https://travis-ci.com/TheThingsNetwork/lorawan-stack/jobs/185199390
By some reason `errors.ErrorDetailsToProto` returns a `proto.Message`: https://github.com/TheThingsNetwork/lorawan-stack/blob/4a1d4fb2e4e5517ed073020f57f61f826dfdcfd9/pkg/errors/grpc.go#L61

Reply:
Shouldn't we change that then?

Reply:
I also think that there's no reason for this abstraction here.
Changing it is out of scope of this PR though, can you create an issue about it?

Reply:
This is not out of scope imo. In continuous improvement, anything that's used in a PR is in scope, also if it's improving the codebase elsewhere.

Reply:
This, however, may have something to do with import paths (?)

Reply:
This is indeed to prevent import cycles. `pkg/errors` is imported by `pkg/ttnpb`, so if a func in `pkg/errors` would return something from `pkg/ttnpb` we have a cycle.

Reply:
How would you advise on this situation?